packages/mds-policy-service/@types/intents.ts

```diff
@@ -0,0 +1,82 @@
+import type { DAY_OF_WEEK, Nullable, Timestamp, UUID } from '@mds-core/mds-types'
+import { now } from '@mds-core/mds-utils'
+import type { BaseRule, MicroMobilityStatesToEvents, TIME_FORMAT } from './models'
+
+export const INTENT_TYPES = ['no_parking'] as const
+export type INTENT_TYPE = typeof INTENT_TYPES[number]
+
+export const BASE_POLICY_DEFAULTS = {
+  name: '',
+  description: '',
+  provider_ids: [],
+  start_date: now(),
+  end_date: null,
+  prev_policies: [],
+  rules: [],
+  currency: null,
+  published_date: null
+}
+
+export const NO_PARKING_INTENT_RULE_FIXED_VALUES: NoParkingIntentRuleDefaults = {
+  geographies: [],
+  days: [],
+  start_time: null,
+  end_time: null,
+  accessibility_options: null,
+  maximum: 0,
+  modality: 'micromobility',
+  rate_amount: null,
+  rate_recurrence: null,
+  name: 'No Parking Zones',
+  rule_type: 'count',
+  rule_units: 'devices',
+  states: { available: [], non_operational: [], reserved: [], unknown: [] },
+  value_url: null,
+  vehicle_types: ['scooter', 'moped'],
+  propulsion_types: [],
+  transaction_types: null,
+  service_types: null
+}
+
+export type NoParkingIntentRuleDefaults = Omit<BaseRule<PartialMicroMobilityStatesMap, 'count'>, 'rule_id'>
+
+export const INTENT_RULE_CONSTANTS: { [K in INTENT_TYPE]: NoParkingIntentRuleDefaults } = {
+  no_parking: NO_PARKING_INTENT_RULE_FIXED_VALUES
+}
+
+export interface BaseIntentPolicyUserFields {
+  name: string
+  description: string
+  provider_ids: Nullable<UUID[]>
+  start_date: Timestamp
+  end_date: Nullable<Timestamp>
+}
+
+export type PartialMicroMobilityStatesMap = Partial<MicroMobilityStatesToEvents>
+export type PartialMicromobilityRule = Omit<Partial<BaseRule<PartialMicroMobilityStatesMap>>, 'rule_id'>
```

Comment:
Are these types necessary? Doesn't look like they're used anywhere

Reply:
The first one is, the second is leftover from experimenting.